rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -44,9 +44,37 @@
 #define HAS_SIGACTION
 #endif
 
+rclcpp::Logger g_logger = rclcpp::get_logger("rclcpp");
+
 namespace rclcpp
 {
 
+// This setups anything that is necessary for wait_for_signal() or notify_signal_handler().
+// This must be called before wait_for_signal() or notify_signal_handler().
+// This is not thread-safe.
+RCLCPP_LOCAL
+void
+setup_wait_for_signal();
+
+// Undo all setup done in setup_wait_for_signal().
+// Must not call wait_for_signal() or notify_signal_handler() after calling this.
+// This is not thread-safe.
+RCLCPP_LOCAL
+void
+teardown_wait_for_signal() noexcept;
+
+// This waits for a notification from notify_signal_handler() in a signal safe way.
+// This is not thread-safe.
+RCLCPP_LOCAL
+void
+wait_for_signal() noexcept;
+
+// This notifies the deferred_signal_handler() thread to wake up in a signal safe way.
+// This is thread-safe.
+RCLCPP_LOCAL
+void
+notify_signal_handler() noexcept;
```

Comment:
@wjwwood consider making above functions private `static` methods of the `SignalHandler` class.

Reply:
I did this in my latest round of changes, but retrospect I feel it's not necessary to stuff everything into the class. I'll leave them as-is though (in the class).